book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+| Release | Post PipelineRun if there is a ReleaseLink for the Application, if the auto-release flag is set on the ReleaseLink | To signal to the Release Service to automatically push the Snapshot to the Production Environment |
+
+### READ
+
+| Custom Resources | When? | Why? |
+|---|---|---|
+| Application & Component | Before creating the Snapshots | To know which build.ImageSpecs to use in the Snapshot |
+| IntegrationTestScenario | Before creating the Component/Composite PipelineRun(s) | To get the information for the PipelineRun - which Tekton bundles and Environments will be used |
+| ReleaseLink | Before creating the Release | To signal the Release Service for next environment promotion (prod) |
+
+### UPDATE
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| SnapshotEnvironmentBinding | Post Test PipelineRun | To promote the Snapshot to the next environment (Stage/Stable) |
+| Snapshot | Post Test PipelineRun | To mark the snapshot validated so that it can be promoted to the next environment |
+| Component | Post Test PipelineRun | To update the Component with the image spec, in order to update the global candidates list |
+
+### WATCH
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| PipelineRun | Post Build PipelineRun | Read the component and the output image to create the Snapshot and start the Preliminary PipelineRun |
+| PipelineRun | Post Preliminary PipelineRun | Check the result of the Preliminary PipelineRun and  follow up on the global candidate list state to either trigger the Final PipelineRun or check ReleaseLink and create a Release |
+| PipelineRun | Post Final PipelineRun | Check the result of the Final PipelineRun, check ReleaseLink and create a Release |
+
+### Pipeline
+
+The test pipeline will consist of a single definition that could be executed as two logically distinct pipeline runs by the Integration Service. The logical pipeline runs are:
+
+- Component (Prelim) pipeline run
+- Composite (Final) pipeline run
+
+The Component pipeline run is always triggered for each Component of an Application and is triggered by a completed build of a component.  
+
+The Composite run is only triggered when it has been determined by the Integration Service that two or more Components of an Application have been successfully built concurrently by their respective Component pipeline and need to be tested together. 
+
+The two stage model is roughly modeled on the 'fast loop / slow loop' model that OpenShift uses.  ART is the slow loop.  The global candidates list at some point in time is the set of artifacts that the Composite pipeline will test and mark as a 'shippable' snapshot.
+
+[Draft: StoneSoup-4 Pipeline & Task Design](https://docs.google.com/document/d/1ucQI3WJzkLpueYLuzHEXpKNvYQafDl4gNBZryvi2d5o/edit#)
+
+### PipelineRun
+
+The Integration Service will generate a PipelineRun that can reference the user provided tekton pipeline. 
```

Comment:
Are the subcategories all properties that the integration will set? Will need access/permissions to? Is any of it user facing/customizable or is it just a description of the service's interface?

Reply:
could you reiterate this question?

Reply:
You have a parent heading with child headings/sections. You do not explicitly describe the relationship between the child headings and a `PipelineRun`. I was trying to suggest ways to tie all the sections together.  If you can have a short description about how the children are related to the parent, that would be great.

For example, are these properties of the PipelineRun that is specified? Are these properties that the integration service controls/required/sets? Are they pre-requisites that the user needs to configure or customize? 

Reading it again, you might be doing a good enough job here for now. Just take another look at it to see if any additional clarification is needed. If you don't think anything else is needed, feel free to resolve this thread.

Reply:
Updated. Done